lib/crypto/scrypt.js

```diff
@@ -56,12 +50,7 @@ module.exports = function(log, config) {
       if (scrypt.numPending > scrypt.numPendingHWM) {
         scrypt.numPendingHWM = scrypt.numPending
       }
-      if (scryptHash) {
-        scryptHash(input, salt, N, r, p, len, (err, hash) => {
-          scrypt.numPending -= 1
-          return err ? d.reject(err) : d.resolve(hash.toString('hex'))
-        })
-      } else if (crypto.scrypt) {
+      if (crypto.scrypt) {
```

Comment:
I think you can delete this `if` and the trailing `else` too, `crypto.scrypt` will always be available.